chpl/chpl-service/src/main/java/gov/healthit/chpl/compliance/directreview/DirectReviewListingSharedStoreHandler.java

```diff
@@ -0,0 +1,120 @@
+        this.sharedListingStoreProvider = sharedListingStoreProvider;
+        this.certifiedProductDetailsManager = certifiedProductDetailsManager;
+        this.listingSearchService = listingSearchService;
+        this.developerDAO = developerDAO;
+    }
+
+    public void handler(List<DirectReview> allDirectReviews) {
+        getUniqueDevelopers(allDirectReviews).stream()
+                .forEach(dev -> getListingDataForDeveloper(dev).stream()
+                        .forEach(listing -> removeListingFromSharedStoreIfDirectReviewUpdated(listing, allDirectReviews)));
+    }
+
+    private List<Developer> getUniqueDevelopers(List<DirectReview> allDirectReviews) {
+        return StreamEx.of(allDirectReviews)
+                .map(dr -> getDeveloper(dr.getDeveloperId()))
+                .filter(dev -> dev != null)
+                .distinct(dev -> dev.getId())
+                .toList();
+    }
+
+    private void removeListingFromSharedStoreIfDirectReviewUpdated(CertifiedProductSearchDetails listing, List<DirectReview> allDirectReviews) {
+        listing.getDirectReviews().parallelStream()
+            .forEach(dr -> {
+                if (hasDirectReviewBeenUpdated(findDirectReview(allDirectReviews, dr.getJiraKey()), dr)) {
+                    LOGGER.info("Removing Listing Id {} from the Shared Store", listing.getId());
+                    sharedListingStoreProvider.remove(listing.getId());
+                } else {
+                    LOGGER.info("Not Removing Listing Id {} from the Shared Store - Direct Review Last Updated Date not changed", listing.getId());
+                }
+            });
+    }
+
+    private Boolean hasDirectReviewBeenUpdated(DirectReview newVersion, DirectReview origVersion) {
+        return !NullSafeEvaluator.eval(() -> newVersion.getLastUpdated(), new Date(Long.MIN_VALUE)).equals(
+                NullSafeEvaluator.eval(() -> origVersion.getLastUpdated(), new Date(Long.MIN_VALUE)));
+
+    }
+
+    private DirectReview findDirectReview(List<DirectReview> directReviews, String jiraKey) {
+        return directReviews.stream()
+                .filter(dr -> dr.getJiraKey().equals(jiraKey))
+                .findAny()
+                .orElse(null);
+    }
+
+    private List<CertifiedProductSearchDetails> getListingDataForDeveloper(Developer developer) {
+        try {
+            //TODO: this needs to switched to use developer id when OCD-4066 gets to STG!!!
+            SearchRequest searchRequest = SearchRequest.builder()
+                    .certificationEditions(Stream.of(CertificationEditionConcept.CERTIFICATION_EDITION_2015.getYear()).collect(Collectors.toSet()))
+                    .developer(developer.getName())
+                    .pageSize(SearchRequest.MAX_PAGE_SIZE)
+                    .build();
+
+            return listingSearchService.getAllPagesOfSearchResults(searchRequest).parallelStream()
+                    .map(searchResult ->  getListing(searchResult.getId()))
```

Comment:
I'm wondering why you need to get the listing details here. I could be missing it, but I suspect the only thing you really need is the ID which you'll use to remove the listing from the shared store. Just trying to avoid extra work if it's not needed. I see the next line filters if the listing is not null. Was there a case where it was null?